src/Components/VideoCard/video-card.js

```diff
@@ -1,18 +1,71 @@
-import '../VideoCard/video-card.css';
+import "../VideoCard/video-card.css";
+import { useNavigate } from "react-router-dom";
+import { useState } from "react";
+import { watchLaterHandler } from "../../Utils/watchLaterUtils";
+import { useAuth } from "../../Context/authContext";
+import { useVideo } from "../../Context/dataContext";
 const VideoCard = ({ video }) => {
-    const {_id,title,catchName,uploaded} = video
+  const [showOption, setShowOption] = useState(false);
+  const navigate = useNavigate();
+  const { token } = useAuth();
+  const { Datadispatch } = useVideo();
+  const { _id, title, catchName, uploaded } = video;
+  const singlePageNavigator = () => {
+    navigate(`/video/${_id}`);
+  };
   return (
     <>
       <div class="container">
         <figure class="menu-card">
-          <img src={`https://i.ytimg.com/vi/${_id}/0.jpg`}/>
+          <img
+            src={`https://i.ytimg.com/vi/${_id}/0.jpg`}
+            onClick={() => singlePageNavigator()}
+          />
           <figcaption>
-            <h3 className='video-title'>{title}</h3>
+            <h3 className="video-title">{title}</h3>
             <p className="subtitle">{catchName}</p>
-            <div className='date-three-bars'>
-            <p>{uploaded}
-            </p>
-            <span><i class="fa-solid fa-ellipsis-vertical"></i></span>
+            <div
+              className="date-three-bars"
+              onClick={() => setShowOption((showOption) => !showOption)}
+            >
+              <p>{uploaded}</p>
+              <span>
+                <i class="fa-solid fa-ellipsis-vertical"></i>
+              </span>
+              <div
+                className={`options ${
+                  showOption ? `display-flex` : `display-none`
+                }`}
+              >
+                <div
+                  className={`${video.isInWatchLater && "red-color"}`}
```

Comment:
in some cases, you are using ternary and in some cases, you are using &, have consistent things